service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AllocatedJudgeControllerMidEventTest.java

```diff
@@ -42,4 +43,45 @@ void shouldReturnAValidationErrorWhenJudgeEmailIsInvalid() {
         assertThat(callbackResponse.getErrors()).contains(
             "Enter an email address in the correct format, for example name@example.com");
     }
+
+
+    @Test
+    void shouldClearLastNameIfMagistrates() {
+        Judge judge = Judge.builder()
+            .judgeEmailAddress("email@example.com")
+            .judgeLastName("lastName")
+            .judgeFullName("fullName")
+            .judgeTitle(JudgeOrMagistrateTitle.MAGISTRATES)
+            .build();
+        CaseData caseData = CaseData.builder()
+            .allocatedJudge(judge)
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseData);
+
+        CaseData caseDataResult = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        assertThat(caseDataResult.getAllocatedJudge())
```

Comment:
may be add another assert that the full name (first name) exists and it is only clearing last name?

Reply:
Those assert are comparing the expected judge and the actual judge instance. The instances are consider equal if all field are equals, including title, last name, full name etc...

The toBuilder() method will keep all the properties and return a builder of the object.